compiler-core/src/javascript/expression.rs

```diff
@@ -295,7 +296,24 @@ impl<'module> Generator<'module> {
                     }
 
                     // Bit arrays
-                    [Opt::Bytes { .. } | Opt::Bits { .. }] => Ok(docvec![value, ".buffer"]),
+                    [Opt::Bits { .. }] => Ok(value),
+
+                    // Bit arrays with explicit size. The explicit size slices the bit array to the
+                    // specified size. A runtime exception is thrown if the size exceeds the number
+                    // of bits in the bit array.
+                    [Opt::Bits { .. }, Opt::Size { value: size, .. }]
+                    | [Opt::Size { value: size, .. }, Opt::Bits { .. }] => match &**size {
+                        TypedExpr::Int { value: size, .. } => {
+                            Ok(docvec![value, ".slice(0, ", size, ")"])
+                        }
+
+                        TypedExpr::Var { name, .. } => Ok(docvec![value, ".slice(0, ", name, ")"]),
+
+                        _ => Err(Error::Unsupported {
+                            feature: "This bit array segment option UNREACHABLE".into(),
```

Comment:
If this is unreachable how come it is here?

Reply:
Sorry, this was leftover from testing and has been reverted as it it is reachable.